homeassistant/components/climate/systemair_savecair.py

```diff
@@ -0,0 +1,249 @@
+"""
+SystemAIR platform for VTR 300 Ventilation Unit.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/savecair/
+"""
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    PLATFORM_SCHEMA, SUPPORT_FAN_MODE, SUPPORT_ON_OFF, SUPPORT_OPERATION_MODE,
+    SUPPORT_TARGET_TEMPERATURE, ClimateDevice)
+from homeassistant.const import (
+    ATTR_TEMPERATURE, CONF_DEVICE, CONF_NAME, CONF_PASSWORD,
+    CONF_SCAN_INTERVAL, TEMP_CELSIUS, TEMP_FAHRENHEIT)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['websockets==6.0', 'python-systemair-savecair==0.0.2']
+
+ATTR_SUPPLY_AIR_TEMPERATURE = "supply_air_temperature"
+ATTR_SUPPLY_AIR_FAN_SPEED = "supply_air_fan_speed"
+ATTR_EXTRACT_AIR_FAN_SPEED = "extract_air_fan_speed"
+
+CONF_IAM_ID = "iam_id"
+
+DEFAULT_NAME = "SystemAIR"
+
+SCAN_INTERVAL = timedelta(seconds=60)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_DEVICE): cv.string,
+    vol.Required(CONF_IAM_ID): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL): cv.time_period,
+})
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the SystemAIR platform."""
+    name = config.get(CONF_NAME)
+    device = config.get(CONF_DEVICE)
+    iam_id = config.get(CONF_IAM_ID)
+    password = config.get(CONF_PASSWORD)
+
+    climate = SystemAIRClimate(name, device, iam_id, password)
+    async_add_entities([climate])
+
+
+class SystemAIRClimate(ClimateDevice):
+    """Representation of the climate sensor."""
+
+    def __init__(self, name, unit, iam_id, password):
+        """Construct the SystemAIR Climate Device."""
+        from systemair.savecair.client import SavecairClient
+        self._name = name
+
+        """Initialize the climate device."""
+        self._support_flags = SUPPORT_TARGET_TEMPERATURE
+
+        if unit == "vtr_300":
+            self._support_flags = self._support_flags | SUPPORT_FAN_MODE
+            self._support_flags = self._support_flags | SUPPORT_OPERATION_MODE
+            self._support_flags = self._support_flags | SUPPORT_ON_OFF
+
+        # Create savecair client
+        self._client = SavecairClient(iam_id, password)
+        self._client.update_cb.append(self.update_callback)
+        self.last_operation_mode = None
+
+        self._fan_list = ['Off', 'Low', 'Normal', 'High']
+        self._operation_list = ['Auto', 'Manual', 'Crowded',
```

Comment:
Operation list must only contain states found in the base climate component. Use two dicts to go from home assistant state to device mode.

Reply:
I've been thinking about this. So basicly, SystemAIR delivers ventilation units for ventilation in homes. Thes have several operation modes, such as "Away, Holiday ...etc" 

The reason I put this component under climate is that it also supports Heating and Cooling in addition to. I do believe that it fits well here because its really just a climate device. I'm not sure on how to approach this issue. Should I suggest additional operations added to HA, or is there a way to show the SystemAIR operations in frontend while using HA operations in the backend:
```
self._operation_list = [STATE_AUTO, STATE_MANUAL, STATE_AUTO, STATE_AUTO, STATE_AUTO, STATE_AUTO, STATE_AUTO]
self._operation_list_map = ['Auto', 'Manual', 'Crowded','Refresh', 'Fireplace', 'Away', 'Holiday']
 ```

Reply:
@MartinHjelmare, do you have any input here? :)

Reply:
There's no way around using the modes in the base climate component. You're welcome to propose a change, but I can't promise that it will be accepted and I don't know how long that could take. We currently have an open issue on climate component restructuring in our architecture repo. Please start by reading that.

Reply:
@MartinHjelmare Maybe, these devices dont belong in the climate component then? These are primarily ventilation units, but there exists about 20 different units. There is 2 ways of interfacing using savecair module or modbus. Would it be possible to have systemair as a base component, or maybe a ventilation base component?

I'm really not sure about what is the most fitting, but i'm quite sure that systemair will be a increasingly used device in newer homes.

Reply:
Why not start by implementing auto, manual and away, which all already exist in the climate component?